### PR TITLE
Issue2

### PR DESCRIPTION
**Pull request by kai-li-27**
Closes #X

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by kai-li-27** [comment]
[Issue 2: Save Image Link](https://trello.com/c/s8nS9o3H/167-issue-2-save-image-link)

File: config/routes.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   root 'application#home'
+  resources :images
```
Comment (eric199wen) [inline config/routes.rb]: Since we will only use several actions in our stories, it might be a good idea to do something like `resources :images, only: [:index, new, create, show]` to only create necessary routes.

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<h1> hello world</h1>
+<a href=<%=new_image_path%>>submit link</a>
```
Comment (eric199wen) [inline app/views/images/index.html.erb]: Although robocop doesn't catch this, we usually like to add a white space after `<%` and before `%>`

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class ImagesControllerTest < ActionDispatch::IntegrationTest
+  test 'Should get images index page' do
+    get images_path
+
+    assert_response :ok
+    assert_select  'h1', 'hello world'
+    assert_select  'a' do
```
Comment (eric199wen) [inline test/controllers/images_controller_test.rb]: It seems like you have an extra whitespace in both line 8 and 9.

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -18,4 +18,19 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+    assert_redirected_to image_path(Image.last)
+  end
+
+  test 'Should create a new image with invalid input' do
+    assert_no_difference('Image.count') do
+      post images_path, params: { image: { url: 'abcd' } }
+    end
+    assert_response :unprocessable_entity
+    assert_select 'form input[name="image[url]"]'
+  end
```
Comment (eric199wen) [inline test/controllers/images_controller_test.rb]: We should also add a test for empty url as well.

File: app/views/images/show.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+<img src=<%=@image.url%>>
```
Comment (eric199wen) [inline app/views/images/show.html.erb]: As I mentioned before, we normally add whitespace after `<%` and before `%>`.

File: config/initializers/simple_form.rb [hunk 6]
```diff
@@ -0,0 +1,182 @@
+# frozen_string_literal: true
```
Comment (ztmeyer) [inline config/initializers/simple_form.rb]: If you hit any rubocop errors in this file, it's also OK to exclude this initializer from the rubocop.yml config 👍

Comment (kai-li-27) [reply config/initializers/simple_form.rb]: So I already fixed the rubocop error for simple form, but I will keep in mind next time. Thanks!

File: app/controllers/images_controller.rb [hunk 7]
```diff
@@ -4,4 +4,18 @@ def index; end
   def new
     @image = Image.new
   end
+
+  def create
+    @image = Image.create(image_params)
```
Comment (eric199wen) [inline app/controllers/images_controller.rb]: `create` combines `new` and `save` together. Here it might be slightly better to do `Image.new(image_params)` first and check if `@image` is valid or not. If image is valid then we save it. 

This way we have some more readability and you won't validate the model twice.

File: test/controllers/images_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class ImagesControllerTest < ActionDispatch::IntegrationTest
+  test 'Should get images index page' do
+    get images_path
+
+    assert_response :ok
+    assert_select 'h1', 'hello world'
+    assert_select 'a' do
+      assert_select '[href=?]', new_image_path
```
Comment (eric199wen) [inline test/controllers/images_controller_test.rb]: We can combine line 9 and 10 together by doing `assert_select a[href=?], new_image_path`.

No need to make any change, just to let you know.
